src/traefik.py

```diff
@@ -162,6 +162,8 @@ def update_cert_configuration(
     ):
         """Update the server cert, ca, and key configuration files."""
         if cert:
+            # write it to the charm container too, for charm tracing.
+            Path(SERVER_CERT_PATH).write_text(cert)
```

Comment:
Looking at itests, this path can be not present in charm container. How about doing `Path.mkdir(DYNAMIC_CONFIG_DIR, parents=True, exist_ok=True)` first?

Reply:
ah yes, good catch